Builds are failing to copy artifacts, Dubious character in TARGET_OS, ARCHITECTURE or VARIANT

We're seeing an error during the copy artifacts stage of all of the dry run builds

```
19:00:29  ERROR: Error during copy and archive artifacts for build job: build-scripts/jobs/release/jobs/jdk17u/jdk17u-release-linux-arm-temurin
19:00:29  Also:   org.jenkinsci.plugins.workflow.actions.ErrorAction$ErrorId: 5d6e732e-7498-48f9-840e-80a12f077405
19:00:29  java.lang.Exception: [ERROR] Dubious character in TARGET_OS, ARCHITECTURE or VARIANT - aborting
```

eg https://ci.adoptium.net/job/build-scripts/job/release-openjdk17-pipeline/50/console

Error can be traced to here:
https://github.com/adoptium/ci-jenkins-pipelines/blob/c58042b6667e06360a56ea80a10d7d3fd7c490d0/pipelines/build/common/build_base_file.groovy#L912

Likely introduced by this pr https://github.com/adoptium/ci-jenkins-pipelines/pull/873

--- a/pipelines/build/common/build_base_file.groovy
+++ b/pipelines/build/common/build_base_file.groovy
@@ -906,9 +906,9 @@ class Builder implements Serializable {
                                         //Remove the previous artifacts
                                         try {
                                             context.timeout(time: pipelineTimeouts.REMOVE_ARTIFACTS_TIMEOUT, unit: 'HOURS') {
-                                                if ( ! ( "${config.TARGET_OS}"    ==~ /^[A-Za-z0-9\/\.-_]$/ ) ||
-                                                     ! ( "${config.ARCHITECTURE}" ==~ /^[A-Za-z0-9\/\.-_]$/ ) ||
-                                                     ! ( "${config.VARIANT}"      ==~ /^[A-Za-z0-9\/\.-_]$/ ) ) {
+                                                if ( ! ( "${config.TARGET_OS}"    ==~ /^[A-Za-z0-9\/\.-_]*$/ ) ||
+                                                     ! ( "${config.ARCHITECTURE}" ==~ /^[A-Za-z0-9\/\.-_]*$/ ) ||
+                                                     ! ( "${config.VARIANT}"      ==~ /^[A-Za-z0-9\/\.-_]*$/ ) ) {
                                                     throw new Exception("[ERROR] Dubious character in TARGET_OS, ARCHITECTURE or VARIANT - aborting");
                                                 }
                                                 context.sh "rm -rf target/${config.TARGET_OS}/${config.ARCHITECTURE}/${config.VARIANT}/"

